src/Facility.CodeGen.CSharp/CSharpServiceInfo.cs

```diff
@@ -81,6 +81,28 @@ public static bool TryCreate(ServiceInfo serviceInfo, CSharpServiceInfoSettings?
 
 	internal string GetErrorName(ServiceErrorInfo errorInfo) => FixName(errorInfo.Name);
 
+	internal string GetExternalDtoName(ServiceExternalDtoInfo info)
+	{
+		var attribute = info.TryGetAttribute("csharp");
+		var typeName = attribute?.TryGetParameterValue("type") ?? $"{FixName(info.Name)}Dto";
+		var typeNamespace = attribute?.TryGetParameterValue("namespace") ?? "";
```

Comment:
~~It would be reasonable to default to the namespace used by this project, in case both definitions use the same namespace.~~ Then again, I suppose the client library could use implicit usings, in which case they might not need a `csharp` attribute at all. So, no change requested.